cibuildwheel/architecture.py

```diff
@@ -46,19 +52,26 @@ def parse_config(config: str, platform: PlatformName) -> 'Set[Architecture]':
     def auto_archs(platform: PlatformName) -> 'Set[Architecture]':
         native_architecture = Architecture(platform_module.machine())
         result = {native_architecture}
+
         if platform == 'linux' and native_architecture == Architecture.x86_64:
             # x86_64 machines can run i686 docker containers
             result.add(Architecture.i686)
+
         if platform == 'windows' and native_architecture == Architecture.AMD64:
             result.add(Architecture.x86)
+
+        if platform == 'macos' and native_architecture == Architecture.arm64:
+            # arm64 can build and test both archs of a universal2 wheel.
```

Comment:
It's unlikely we'll be able to leverage this code path anytime soon with the lack of arm64 CI runners but should `x86_64` be added here ? Or is that on purpose and only do `x86_64` on `x86_64` runners with `auto` ?

Reply:
Good point, it is on purpose but maybe I haven't explained my thought process fully. Once arm64 runners are commonplace, I'm guessing that the best setup would be to have two macOS runners - one x86_64, and one arm64. This is so we test natively as much as possible, and because it's _probably_ better to build all of cp35-cp38/pp on the x86_64 runner.

So, to avoid doubling-up of wheels, the arm64 runner doesn't build x86_64 by default. 

Ultimately, users would upgrade cibuildwheel and add an arm64 runner in their matrix, and the defaults do all the right things.